internals-js/src/types.ts

```diff
@@ -32,26 +30,25 @@ export type SubtypingRule = typeof ALL_SUBTYPING_RULES[number];
 export const DEFAULT_SUBTYPING_RULES = ALL_SUBTYPING_RULES.filter(r => r !== "list_upgrade");
 
 /**
- * Tests whether 2 types are the same type.
+ * Tests whether 2 types are the "same" type.
  *
- * To be the same type, for this method, is defined as having the same "kind"
- * and either the same name, for named types or, for wrapper types, the same
- * wrapped type (applying this method recursively).
+ * To be the same type, for this method, is defined as having the samee name for named types
+ * or, for wrapper types, the same wrapper type and recursively same wrapped one.
  *
- * This method does not check that both types are from the same schema and does
- * not validate that the structure of named types is the same.
+ * This method does not check that both types are from the same schema and does not validate
+ * that the structure of named types is the same. Also note that it does not check the "kind"
+ * of the type, which is actually relied on due to @interfaceObject (where the "same" type 
+ * can be an interface in one subgraph but an object type in another, while fundamentally being
+ * the same type).
  */
 export function sameType(t1: Type, t2: Type): boolean {
```

Comment:
The `kind` was previously just used as a short-circuit to not have to recurse, but this won't change any existing behavior in theory, correct?

Reply:
I honestly didn't consider when writing this that comparing type of different kind could ever matter, so yes that was mostly a short-circuit to ensure the recursion worked. But yes, I reviewed the call site of this and I'm almost certain this won't change any behavior in an intended way (I mean, if we compare types from the same subgraph, this change makes no difference since a given name uniquely identify a type object in a given schema; so this only matter where comparing types from different schema where we check the kind of types between subgraph independently anyway, so this can't really allow things we don't want).